samples/ClientSample/RawSample.cs

```diff
@@ -78,5 +68,37 @@ internal static void Register(CommandLineApplication app)
             }
             return 0;
         }
+
+        private static async Task ReceiveLoop(TextWriter output, PipeReader input)
+        {
+            while (true)
+            {
+                var result = await input.ReadAsync();
+                var buffer = result.Buffer;
+
+                try
+                {
+                    if (result.IsCanceled || (buffer.IsEmpty && result.IsCompleted))
```

Comment:
Copy the server side loop pattern:

if (!buffer.IsEmpty)
{
     // Do stuff with buffer
}
else if (result.IsCompleted)
{
    break;
}